crates/control-interface/src/lib.rs

```diff
@@ -674,6 +709,51 @@ pub async fn collect_sub_timeout<T: DeserializeOwned>(
     items
 }
 
+enum IdentifierKind {
+    HostId,
+    ActorId,
+    ActorRef,
+    ProviderId,
+    ProviderRef,
+    ContractId,
+    LinkName,
+}
+
+//NOTE(ahmedtadde): For an initial implementation, we just want to make sure that the identifier is, at very least, not an empty string.
+//This parser should be refined over time as needed.
+fn parse_identifier<T: AsRef<str>>(kind: &IdentifierKind, value: T) -> Result<String> {
+    let value = value.as_ref();
+    let assert_non_empty_string = |input: &str, message: Option<String>| -> Result<String> {
+        if input.trim().is_empty() {
+            Err(format!(
+                "Empty string provided for {}",
+                message.unwrap_or_else(|| "entity identifier".to_string())
+            )
+            .into())
+        } else {
+            Ok(input.trim().to_string())
```

Comment:
Might be better to just have this return a unit type instead of the string. In the future, we might add additional verifications, which means we should let the caller decide when to clean up the input and convert it to a string